common/src/main/java/com/wynntils/mc/event/ServerResourcePackLoadEvent.java

```diff
@@ -8,15 +8,17 @@
 import net.minecraftforge.eventbus.api.Cancelable;
 import net.minecraftforge.eventbus.api.Event;
 
-/** Fires on receiving {@link net.minecraft.network.protocol.game.ClientboundResourcePackPacket} */
-@EventThread(EventThread.Type.IO)
+/**
+ * Fired on receiving {@link net.minecraft.network.protocol.common.ClientboundResourcePackPacket}
+ */
+@EventThread(EventThread.Type.ANY)
 @Cancelable
-public class ResourcePackEvent extends Event {
+public class ServerResourcePackLoadEvent extends Event {
```

Comment:
Maybe you could combine the two under a ServerResourcePackEvent with a Load and Clear nested subclass? Just to keep down on the number of event files.